File: 2020/2020W02/PreppindataW02.py
import pandas as pd
import re

def format_time(tstr):
    return str(int(re.search(r'(\d+)(?=\d{2})',tstr).group(1))+
               (12 if  re.search(r'(p|P)',tstr) else 0)).zfill(2) + ':' + re.search(r'(\d{2})(?=\D|$)',tstr).group(1)

final = pd.read_csv(r'E:/PD 2020 Wk 2 Input - Time Inputs.csv' )
final['Time'] = final['Time'].apply(lambda x: format_time(re.sub('\s|\W','',x)))
final['Date Time'] = pd.to_datetime(final['Date'] + ' ' + final['Time'], format='%m/%d/%y %H:%M')
final['Date'] = final['Date Time'].dt.date

File: 2020/2020W02/preppindataw02.R
library(dplyr)
library(readr)
library(stringr)

final <- read_csv("E:/PD 2020 Wk 2 Input - Time Inputs.csv", col_types = 
                    cols(Date = col_date(format = "%m/%d/%y"), 
                         Time = col_character())) %>%
  mutate('Temp' = str_remove(`Time`,'\\s|\\W'),
         'Time' =  paste0(str_pad(as.integer(str_extract(`Temp`,'(\\d+)(?=\\d{2})'))
                                  +if_else(str_detect(`Temp`,'(p|P)'),12,0),
                                  2,'left','0'),
                          ':',
                          str_extract(`Temp`,'(\\d{2})(?=\\D|$)')),
         'Date Time' = as.POSIXct(paste0(format(`Date`,'%Y/%m/%d'),' ',`Time`), format = "%Y/%m/%d %H:%M")) %>%
  select(-'Temp')

View(final)